content/blog/2021-07-13-hyperparam-tuning.md

```diff
@@ -0,0 +1,295 @@
+│ ├── ec60206 [exp-5df5f] │  0.55315 │ 0.93795 │ 8               │ 400         │
+│ ├── 367d0fc [exp-f9edb] │  0.55135 │ 0.93455 │ 16              │ 400         │
+│ ├── 45a2ee6 [exp-24f26] │  0.54877 │ 0.93467 │ 32              │ 400         │
+│ ├── 967a9c2 [exp-67a0c] │  0.53505 │ 0.93703 │ 64              │ 400         │
+│ ├── bccfed8 [exp-83d3e] │  0.52921 │ 0.93681 │ 128             │ 400         │
+│ ├── a3e4182 [exp-9ecac] │  0.52573 │ 0.93606 │ 256             │ 400         │
+│ ├── 6b8424c [exp-42ba9] │  0.55812 │ 0.93587 │ 8               │ 450         │
+│ ├── d075700 [exp-d1a08] │  0.55283 │ 0.93396 │ 16              │ 450         │
+│ ├── ab1e718 [exp-3ad29] │  0.54889 │ 0.93361 │ 32              │ 450         │
+│ ├── 35cb92c [exp-4edb8] │  0.53681 │ 0.93608 │ 64              │ 450         │
+│ ├── f0f44e8 [exp-dc6c7] │  0.53642 │ 0.93525 │ 128             │ 450         │
+```
+
+Now you can see how your precision changed with each hyperparameter value
+update. This is a quick implementation of grid search in DVC. You could read the
+hyperparameter values from a different file or data source or make this tuning
+script as fancy as you like. The main thing you need is the
+`dvc exp run --queue --set-param <param>` command to execute when you add new
+values.
+
+### Random search
+
+Another commonly used method for tuning hyperparameters is random search. This
+takes random values for hyperparameters and builds the model with them. It
+usually takes less time than an exhaustive grid search and it can perform better
+if run for a similar amount of time as a grid search.
+
+We're going to add a example of random search in a new file called
+`random_search.py` simialr to the file we created for grid search. This will add
+queued experiments with the randomly selected hyperparameter values. Add the
+following code to `random_search.py`.
+
+```python
+import subprocess
+import random
+
+# Automated random search experiments
+num_exps = 10
+random.seed(0)
+
+for _ in range(num_exps):
+    params = {
+        "rand_n_est_value": random.randint(250, 500),
+        "rand_min_split_value": random.choice([8, 16, 32, 64, 128, 256])
+    }
+    subprocess.run(["dvc", "exp", "run", "--queue",
+                    "--set-params", f"train.n_est={params['rand_n_est_value']}",
+                    "--set-params", f"train.min_split={params['rand_min_split_value']}"])
+```
+
+This search could be far more complex with Bayesian optimization to handle the
+hyperparameter value selections, but we're keeping it super simple by choosing
+random numbers to focus on reproducibility. This will generate ten experiments
+with random values for each hyperparameter.
+
+You can run these new experiments with `dvc exp run --run-all` and then take a
+look at the results with
+`dvc exp show --include-params=train.min_split,train.n_est --no-timestamp`. Your
+table should look something like this.
+
+```dvc
```

Comment:
Can you update the table with the new results now that you aren't using `train.min_split = 8192`?